### PR TITLE
pgwire: test that pgwire-level prepares fail the txn

### LINKED ISSUE #21495: sql/pgwire: pgwire-level failed prepares do not fail the current txn
Found by @dt: 

- a sql-level PREPARE statement trashes the current txn if it fails, as it should.
- a pgwire-level prepare messages does not trash the current txn if it fails. Pretty sure we never really thought about that. Is this expected? What does pg do?

### PR DESCRIPTION
**Pull request by jordanlewis**
Closes #21495.

Release note: None

### CODE REVIEW

File: pkg/sql/pgwire/pgwire_test.go [hunk 1]
```diff
@@ -2311,3 +2311,35 @@ func TestCancelRequest(t *testing.T) {
...
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	tx, err := db.Begin()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = tx.Prepare("select fail")
```
Comment (dt) [inline pkg/sql/pgwire/pgwire_test.go]: just curious: why not `if _, err :=..` here/below?

**Review by dt — APPROVED** [review]

File: pkg/sql/pgwire/pgwire_test.go (hunk 1 above)
Comment (jordanlewis) [reply pkg/sql/pgwire/pgwire_test.go]: no reason, just didn't do it - done now

**Comment by jordanlewis** [comment]
TFTR!

bors r+

**Pull request merged**